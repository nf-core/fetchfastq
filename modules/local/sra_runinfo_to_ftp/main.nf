
process SRA_RUNINFO_TO_FTP {

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    path runinfo

    output:
    path "*.tsv"       , emit: tsv
    tuple val("${task.process}"), val('python'), eval("python --version | sed 's/Python //g'"), topic: versions

    script:
    """
    sra_runinfo_to_ftp.py \\
        ${runinfo.join(',')} \\
        ${runinfo.toString().tokenize(".")[0]}.runinfo_ftp.tsv
    """
}
