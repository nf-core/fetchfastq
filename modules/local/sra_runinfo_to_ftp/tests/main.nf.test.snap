{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e"
                ],
                "1": [
                    "versions.yml:md5,e95f8185f665127a73622a19d321bcca"
                ],
                "tsv": [
                    "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e"
                ],
                "versions": [
                    "versions.yml:md5,e95f8185f665127a73622a19d321bcca"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-17T09:17:23.371547"
    }
}